dspace-api/src/main/java/org/dspace/discovery/SolrServiceImpl.java

```diff
@@ -425,23 +464,42 @@ public void updateIndex(Context context) {
      */
     @Override
     public void updateIndex(Context context, boolean force) {
-        try {
-            Iterator<Item> items = null;
-            for (items = itemService.findAllUnfiltered(context); items.hasNext(); ) {
-                Item item = items.next();
-                indexContent(context, item, force);
-                //To prevent memory issues, discard an object from the cache after processing
-                context.uncacheEntity(item);
-            }
+        updateIndex(context, force, Constants.ITEM);
+        updateIndex(context, force, Constants.COLLECTION);
+        updateIndex(context, force, Constants.COMMUNITY);
+    }
 
-            List<Collection> collections = collectionService.findAll(context);
-            for (Collection collection : collections) {
-                indexContent(context, collection, force);
-            }
 
-            List<Community> communities = communityService.findAll(context);
-            for (Community community : communities) {
-                indexContent(context, community, force);
+    @Override
+    public void updateIndex(Context context, boolean force, int type) {
+        try {
+            switch (type) {
+                case Constants.ITEM:
+                    Iterator<Item> items = itemService.findAllUnfiltered(context);
+                    for (Item item : ImmutableList.copyOf(items)) {
+                        indexContent(context, item, force);
```

Comment:
I think it's **very important** here that we keep the `context.uncacheEntity(item)` here. Otherwise, Hibernate will keep all items in-memory in the current Hibernate session and this will most certainly lead to memory issues in large repositories.

Reply:
done